intelmq/lib/message.py

```diff
@@ -106,14 +106,47 @@ def __init__(self, message=(), auto=False, harmonization=None):
     def __setitem__(self, key, value):
         self.add(key, value)
 
-    def add(self, key, value, sanitize=True, force=False, ignore=()):
-        if not force and key in self:
-            raise exceptions.KeyExists(key)
+    def is_valid(self, key, value, sanitize=True):
+        """
+        Checks if a value is valid for the key.
 
-        if value is None or value == "":
-            return
+        Parameters
+        ==========
+        key : string
+        value : string
+        sanitize : boolean
+
+        Returns
+        =======
+        retval : boolean
+
+        Raises
+        ======
+        intelmq.lib.exceptions.InvalidKey if key is invalid
+
+        """
+        if not self.__is_valid_key(key):
+            raise exceptions.InvalidKey(key)
+
+        if value is None or value in ["", "-", "N/A"]:
+            return False
+        if sanitize:
+            value = self.__sanitize_value(key, value)
+        valid = self.__is_valid_value(key, value)
+        if valid[0]:
+            return True
+        return False
 
-        if value in ["-", "N/A"]:
+    def add(self, key, value, sanitize=True, force=False, overwrite=False, ignore=(),
+            raise_failure=True):
```

Comment:
**Suggestion:** Add the same kind of documentation as you have in `is_valid` method with special attention to `raise_failure` parameter.